src/core/AnimatedFunction.js

```diff
@@ -0,0 +1,28 @@
+import AnimatedNode from './AnimatedNode';
+import { createAnimatedCallFunc } from './AnimatedCallFunc';
+
+class AnimatedFunction extends AnimatedNode {
```

Comment:
How to make it more "reactive"? I'm a bit afraid about this part, if it's not forced to be in global scope. Maybe we can make it a hook:
```js
useEffect(() => {
  ...
  a.attach()
  return a.detach
}, [inputs] 
)
```

Reply:
Great that you understand the idea :) The only nodes that will stay attached are the function nodes - they should not need to be detached, should they? 

I agree with the `proc((a, b) => multiply(a,b))` API - it looks super nice! I'll try to see if I can implement that one! Don't think typing will be to hard though.

Reply:
I think you are right with the hook idea - it should definitely be used when functions are defined inside components to avoid memory leaks.

Reply:
`function proc(cb: (...args: any[]) => AnimatedNode<any>): AnimatedNode<any>;`

Reply:
Then we can ditch the `param` function from the js side :)

Reply:
I've added the change in API you suggested, and also the `useProc` hook. I'm a little bit uncertain about how we can solve attaching functions without hooks. 

I would imagine that you'd often define a function in the constructor of a component or as a global in a Javascript file and are struggling a bit with how we can ensure that these are registered correctly.

Reply:
@chrfalch according to what @terrysahaidak described as his use case:

```javascript
// Consider this as a meta code, describing a use case(or desired use case)
import { TextInput } from 'reanimated'

let myFunc = proc((x, y) => add(x, y)) 

const Input = (props) => { return ( <TextInput />) } 

Input.initAnimation = () => myFunc.__attach()

// Use this in a case like: do this on a screen A,
// but before going to the screen B, you would use Input.initAnimation().
// While screen be is importing and using this Input.
```
So I kind of wrote something like this:
```javascript
const test = (x, y) => multiply(x, y)
let myStuff = createAnimatedFunction(test)
// Somewhere
myStuff.__attach()
// Elsewere
myStuff.__detach()
```
And this kind of code should probably work because `AnimatedNode` has `__attach` and `__detach`. The only difference to attaching you've done is that should not require/force us to be within React component lifecycle to run this.

Question is: will they be registered as nodes via Reanimated properly. Otherwise good job and thank you!

Reply:
@chrfalch according to what @terrysahaidak described as his use case:
```javascript
// Consider this as a meta code, describing a use case(or desired use case)
import { TextInput } from 'reanimated'

let myFunc = proc((x, y) => add(x, y)) 

const Input = (props) => { return ( <TextInput />) } 

Input.initAnimation = () => myFunc.__attach()

// Use this in a case like: write your component on a screen A(or even elsewhere),
// but before going to the screen B, you would use Input.initAnimation().
// While screen be is importing and using this Input.
```
So I kind of wrote something like this:
```javascript
const test = (x, y) => multiply(x, y)
let myStuff = createAnimatedFunction(test)
// Somewhere
myStuff.__attach()
// Elsewere
myStuff.__detach()
```
And this kind of code should probably work because AnimatedNode has `__attach` and `__detach`. The only difference to attaching you've done in `useProc` is that it should not require/force us to be within React component lifecycle to do this.

Question is: will they be registered as nodes via Reanimated properly. Otherwise good job and thank you!

Reply:
I've added some logic to the createAnimatedFunction (use the exported `proc` and you get it automatically) that attaches automatically when outside the `useProc` hook. When called using `useProc` hook attach/detach will be run automatically for you. It is therefore not recommended to store proc nodes as class members. Make them global instead:

```js
let myFunc

class MyComp extends React.Component {
  constructor(props){
    super(props);
    if(!myFunc) {
      myFunc = proc((a,b) => add(a,b);
    }
  }
}
```

Reply:
Thank you @chrfalch. What looks really promising. The only change (feature request?) I would suggest is to add an ability to `await` proc attaching. My use case is the screen with a big form of identical inputs. And if I know the user is gonna navigate to that screen. I would initialize that proc and wait until it has been initialized and only then navigate to that specific screen, or just won't show the form for the user until. So the user will exactly know the app works as expected and there are no lags. The snippet above probably won't work because we might render 5 identical components at once, so there is a big chance to create that proc several times.

Reply:
Oh, yes, since proc will return something synchronously, makes sense. It's time to take a break :D